### PR TITLE
[SPARK-35392][ML][PYTHON] Fix flaky tests in ml/clustering.py and ml/feature.py 

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?

This PR removes the check of `summary.logLikelihood` in  ml/clustering.py - this GMM test is quite flaky. It fails easily e.g., if:
- change number of partitions;
- just change the way to compute the sum of weights;
- change the underlying BLAS impl

Also uses more permissive precision on `Word2Vec` test case.

### Why are the changes needed?

To recover the build and tests.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing test cases.

### CODE REVIEW

**Comment by zhengruifeng** [comment]
ping @HyukjinKwon @srowen @viirya

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Review by dongjoon-hyun — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@zhengruifeng would you mind fixing:

```
**********************************************************************
File "/__w/spark/spark/python/pyspark/ml/feature.py", line 4681, in __main__.Word2Vec
Failed example:
    model.getVectors().show()
Expected:
    +----+--------------------+
    |word|              vector|
    +----+--------------------+
    |   a|[0.09511678665876...|
    |   b|[-1.2028766870498...|
    |   c|[0.30153277516365...|
    +----+--------------------+
    ...
Got:
    +----+--------------------+
    |word|              vector|
    +----+--------------------+
    |   a|[0.09511695802211...|
    |   b|[-1.2028766870498...|
    |   c|[0.30153274536132...|
    +----+--------------------+
    <BLANKLINE>
**********************************************************************
```

too? feel free to change the JIRA.

I think we can just fix it like:

```
    +----+--------------------+
    |word|              vector|
    +----+--------------------+
    |   a|[0.0951 ...
    |   b|[-1.202 ...
    |   c|[0.3015 ...
    +----+--------------------+
```

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -4682,9 +4682,9 @@ class Word2Vec(JavaEstimator, _Word2VecParams, JavaMLReadable, JavaMLWritable):
     +----+--------------------+
     |word|              vector|
     +----+--------------------+
-    |   a|[0.09511678665876...|
-    |   b|[-1.2028766870498...|
-    |   c|[0.30153277516365...|
+    |   a|[0.0951 ...
+    |   b|[-1.202 ...
+    |   c|[0.3015 ...
```
Comment (HyukjinKwon) [inline python/pyspark/ml/feature.py:4687]:
```suggestion
    |   a|[0.0951...
    |   b|[-1.202...
    |   c|[0.3015...
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
Thanks @zhengruifeng for fixing this!

**Comment by dongjoon-hyun** [comment]
Yes, one stone for two birds! Nice!

**Pull request closed without merging**